app/install.php

```diff
@@ -362,9 +362,26 @@ function printStep0() {
 <?php
 }
 
+function getProcessUsername() {
+	if (function_exists('posix_getpwuid') && function_exists('posix_geteuid')) {
+		$processUser = posix_getpwuid(posix_geteuid());
+		return $processUser['name'];
+	}
+
+	if (function_exists('exec')) {
+		exec('whoami', $output);
+		if (!empty($output)) {
```

Comment:
```suggestion
		if (!empty($output[0])) {
```

My previous suggestion + comment were erased by a forced push, so here they are again, just to finish the conversation.

Does not matter in practice here, but I would still argue that the correct test is `!empty($output[0])`

Three examples:

```php
$output = null;
empty($output[0]); //return true - what we want

$output = [ 1 => 'Test' ];
empty($output); //return false - not what we want
empty($output[0]); //return true - what we want

$output = [ '' ];
empty($output); //return false - not what we want
empty($output[0]); //return true - what we want
```

Reply:
fixed